apps/aecore/src/aec_consensus_hc.erl

```diff
@@ -292,40 +294,84 @@ state_pre_transform_node(Type, Height, PrevNode, Trees) ->
     {ok, PrevHash} = aec_headers:hash_header(PrevHeader),
     {TxEnv0, _} = aetx_env:tx_env_and_trees_from_hash(aetx_transaction, PrevHash),
     TxEnv = aetx_env:set_height(TxEnv0, Height),
+
     {ok, #{epoch := Epoch, first := EpochFirst, last := EpochLast} = EpochInfo} =
         aec_chain_hc:epoch_info({TxEnv, Trees}),
-    {ok, Leader} = leader_for_height(Height, {TxEnv, Trees}),
+
+    #{timeslot := Timeslot} = get_timeslot_and_epoch({TxEnv, Trees}),
```

Comment:
Wait, what? Timeslot is computed using wallclock time (Now)? How can this work when syncing? 🤔

Reply:
Okay, sounds like we need a different source of time, which will correctly point to the chain time when syncing, and which will be wall clock when producing, i.e. discriminate between two modes of operation. I might need more reading on how sync mechanism works.

Reply:
For syncing i use height from the epoch which is equivalent of timeslots if no slot was skipped, so that should work.